eng/common/scripts/modules/ChangeLog-Operations.psm1

```diff
@@ -0,0 +1,127 @@
+# Common Changelog Operations
+
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+.*(?<version>\b\d+\.\d+\.\d+([^0-9\s][^\s:]+)?)(\s(?<releaseStatus>\(Unreleased\)|\(\d{4}-\d{2}-\d{2}\)))?)"
+
+# Returns a Collection of ReleaseNotes object containing changelog info for all version present in the gived CHANGELOG
+function Get-ReleaseNotes {
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]$ChangeLogLocation
+    )
+    
+    $ErrorActionPreference = 'Stop'
+
+    $releaseNotes = @{}
+    if ($ChangeLogLocation.Length -eq 0)
+    {
+        return $releaseNotes
+    }
+    
+    try 
+    {
+        $contents = Get-Content $ChangeLogLocation
+        # walk the document, finding where the version specifiers are and creating lists
+        $releaseNotesEntry = $null
+        foreach($line in $contents){
+            if ($line -match $RELEASE_TITLE_REGEX)
+            {
+               $releaseNotesEntry = [pscustomobject]@{ 
+                  ReleaseVersion = $matches["version"]
+                  ReleaseStatus = $matches["releaseStatus"]
+                  ReleaseTitle = $line
+                  ReleaseContent = @() # Release content without the version title
+               }
+               $releaseNotes[$releaseNotesEntry.ReleaseVersion] = $releaseNotesEntry
+            }
+            else
+            {
```

Comment:
The indention and and spacing is inconsistent in this file. Can you please convert the indentions to be 2 spaces and reformat this entire file?